tests/TestCase/Network/ResponseTest.php

```diff
@@ -2778,6 +2778,49 @@ public function testWithBody()
     }
 
     /**
+     * Test with string body.
+     *
+     * @return void
+     */
+    public function testWithStringBody()
+    {
+        $response = new Response();
+        $response->withStringBody('Foo');
+        $body = $response->getBody();
+        $body->rewind();
+        $result = $body->getContents();
+        $this->assertSame('Foo', $result);
```

Comment:
String casting a stream should rewind and get the contents. That should help make the tests simpler.